src/components/download-mushaf-button.tsx

```
@@ -0,0 +1,345 @@
+                            }
+                            sizesRef.current[item.link].loaded = data.loaded;
+                            sizesRef.current[item.link].total = data.total;
+
+                            // Throttle React state updates to ~every 200ms
+                            const now = Date.now();
+                            if (now - lastRenderTime > 200) {
+                                lastRenderTime = now;
+                                updateOverallProgress();
+                            }
+                        } else if (data.success) {
+                            if (!sizesRef.current[item.link]) {
```

ESLint

[error] Delete `··············`

(prettier/prettier)